skrf/calibration/calibration.py

```diff
@@ -1993,12 +1998,32 @@ def __init__(self, measured, ideals, switch_terms=None,
         switch_terms : tuple of :class:`~skrf.network.Network` objects
             the pair of switch terms in the order (forward, reverse)
 
+        isolation : :class:`~skrf.network.Network` object
+            Measurement with loads on both ports with a perfect isolation
+            between the ports. Used for determining the isolation error terms.
+            If no measurement is given leakage is assumed to be zero.
+
+            Loads don't need to be same as the one used as a match standard or
+            even have similar reflection coefficients. Reflects can be also used,
+            but accuracy might not be as good.
 
         '''
 
         self.switch_terms = switch_terms
         if switch_terms is None:
             warn('No switch terms provided')
+
+        if isolation is None:
+            self.isolation = measured[0].copy()
+            self.isolation.s[:,:,:] = 0
+        else:
+            self.isolation = isolation.copy()
+
+            measured = [m.copy() for m in measured]
+            for i in range(len(measured)):
+                measured[i].s[:,1,0] -= self.isolation.s[:,1,0]
+                measured[i].s[:,0,1] -= self.isolation.s[:,0,1]
+
```

Comment:
this seems to correct for isolation during the init, which i think is unexpected. seems more logical that isolation is corrected for in `apply_cal`. also, i would not expect the `measured` to change. 

maybe we could make a property  `measured_unisolated` similar to `measured_terminated`. then `measured_unterminated` would `unterminate(measured_unisolated)`